crates/iroha_core/src/sumeragi/message.rs

```diff
@@ -39,17 +42,12 @@ impl ControlFlowMessage {
 /// `BlockCreated` message structure.
 #[derive(Debug, Clone, Decode, Encode)]
 pub struct BlockCreated {
+    /// The leaders peer id. Helps for cheap pruning of messages from previous rounds.
+    pub leader_signature: SignatureOf<Vec<SignedTransaction>>,
     /// The corresponding block.
-    pub block: SignedBlock,
```

Comment:
There is no need to change this. Leader can keep sending `SignedBlock` as before

Reply:
I say this because transaction validation will not change anything except modify the `CommittedTransaction::error` value

Reply:
IMO leader can send a block where `CommittedTransaction::error` is `None` and update the value later during block validation